crates/dbsp/src/trace/spine_fueled.rs

```diff
@@ -91,19 +91,64 @@ use crate::{
     trace::{
         cursor::CursorList, Batch, BatchReader, BatchReaderFactories, Cursor, Filter, Merger, Trace,
     },
-    NumEntries,
+    Error, NumEntries, Runtime,
 };
 
 use crate::dynamic::ClonableTrait;
 use rand::Rng;
 use rkyv::{ser::Serializer, Archive, Archived, Deserialize, Fallible, Serialize};
 use size_of::SizeOf;
+use std::fs::File;
+use std::io::Write as IoWrite;
+use std::path::{Path, PathBuf};
 use std::{
     fmt::{self, Debug, Display, Formatter, Write},
+    fs,
     mem::replace,
     ops::DerefMut,
 };
 use textwrap::indent;
+use uuid::Uuid;
+
+/// A spine that is serialized to file.
+#[derive(rkyv::Serialize, rkyv::Deserialize, rkyv::Archive)]
+struct CommittedSpine<B: Batch> {
+    batches: Vec<String>,
+    merged: Vec<(String, String)>,
+    lower: Vec<B::Time>,
+    upper: Vec<B::Time>,
+    effort: u64,
+    dirty: bool,
+    //lower_key_bound: Option<Box<B::Key>>,
+    //key_filter: Option<B::Key>,
+    //value_filter: Option<B::Val>,
+}
+
+impl<B: Batch> From<&Spine<B>> for CommittedSpine<B> {
+    fn from(value: &Spine<B>) -> Self {
+        let mut batches = vec![];
+        value.map_batches(|b| {
+            batches.push(
+                b.persistent_id()
+                    .expect("Persistent spine needs an identifier")
+                    .to_string_lossy()
+                    .to_string(),
```

Comment:
This looks redundant but I didn't look closer.

Reply:
Unfortunately, to_string_lossy doesn't return a string but a Cow<str>